### PR TITLE
TLAPS solution

### PR DESCRIPTION
**Pull request by lorin**
*"The marvel is not that the bear dances well, but that the bear dances at all."*
— Russian proverb

A Hillel Wayne production without a TLA+ solution!? That had to be remedied.

This is a 500 line proof for a 5 line program. This proof could be significantly simplified, especially in the hands of someone with more TLAPS experience than I. But, damn it, the checker says it's correct, and when that happened, I shipped it.

### CODE REVIEW

File: tlaps/readme.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+Invariants: Correct
+Definition Override:
+    Nat <- 0..3
+    Seq(S) <- UNION {[1..m -> S] : m \in Nat}
+```
+
+In my case, this model was small enough to run quickly and large enough to
+find all of the problems with my proposed inductive invariants.
+
+In general, checking an inductive invariant is hard because the state space can
```
Comment (hwayne) [inline tlaps/readme.md]: To clarify, is this saying that it's only proved correct here where the alphabet consists of two symbols and the maximum string length is 3?

Comment (lorin) [reply tlaps/readme.md]: No, the proof is general: it holds for alphabets of any size and strings of any (finite) length.

This text just describes the process I went through to find an inductive invariant. If it turned out that `Inv` was not an inductive invariant, then I wouldn't have been able to write a correct proof, and any attempt to write a proof would have failed the TLAPS check.

Comment (lorin) [reply tlaps/readme.md]: As a side note, TLAPS is completely independent of TLC: you don't need to define a TLC model in order to check a proof with TLAPS.

Comment (lorin) [reply tlaps/readme.md]: I added some text to the readme to try and make it clearer that the proof holds for the general case.

**Comment by hwayne** [comment]
lgtm now. Did you want to add a bit about yourself or would you like me to merge as is?

**Comment by lorin** [comment]
I add an "about me" section, ready to merge!

**Comment by hwayne** [comment]
Hurrah!

**Pull request merged**